tests/pytests/test_multithread.py

```diff
@@ -517,3 +517,59 @@ def check_threads(expected_num_threads_alive, expected_n_threads):
     # Query should not be executed by the threadpool
     env.expect('ft.search', 'idx', '*').equal([0])
     env.assertEqual(getWorkersThpoolStats(env)['totalJobsDone'], 1)
+
+
+def testNameLoader(env: Env):
+    def get_RP_name(profile_res):
+        shard0 = to_dict(profile_res[1])['Shards'][0]
+        last_rp = to_dict(shard0)['Result processors profile'][-1]
+        return to_dict(last_rp)['Type']
+
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'sortable', 'TEXT', 'SORTABLE', 'UNF', 'not-sortable', 'TEXT').ok()
+    with env.getClusterConnectionIfNeeded() as con:
+        for i in range(10):
+            con.execute_command('HSET', f'doc:{i}', 'sortable', f'S{i}', 'not-sortable', f'NS{i}')
+
+    normal_search = env.cmd('FT.SEARCH', 'idx', '*', 'SORTBY', 'sortable', 'RETURN', 1, '__key')
+    normal_aggregate = env.cmd('FT.AGGREGATE', 'idx', '*', 'SORTBY', '1', '@sortable', 'LOAD', 3, '@__key', 'AS', 'doc_id')
+
+    # enable unstable features so we have the special loader
+    verify_command_OK_on_all_shards(env, config_cmd(), 'SET', 'ENABLE_UNSTABLE_FEATURES', 'true')
```

Comment:
Will be needed on the version branches, not here

Reply:
Not needed on `master`? I saw this option is off by default, no?

Reply:
I'm going to keep this feature under the `UNSTABLE_FEATURES` flag even on `master` and `8.0`, so we can implement it better in the future